### PR TITLE
[feat]: Add ability for external party to submit questionnaire reply

### PR DESCRIPTION
**Pull request by MauritsR**
Part 2 of https://github.com/Signalen/product-steering/issues/286. For part 1, see https://github.com/Amsterdam/signals-frontend/pull/2414. 

### Context

This PR adds a new form allowing an external party to see what actions they should take and reply when they are ready.  

After an incident is forwarded to an external party, the external party receives an e-mail with a link to the _Signalen_ application. A questionnaire is shown with an explanation (title, location of the incident, attachments of the incident) and a form (text area, image upload). The backend configures what questions are shown. Currently the form only supports plain text and image answers. Upon completion of the form, the answer is submitted and the backend locks the form. 

### Changes

- Move AttachmentViewer to shared folder and change the attachment type
- Add prop to MapDetail to prevent marker from being focused
- Add new route to ExternalReplyContainer component
- Move IncidentReplyContainer components to ReplyForm folder (Notice, FileInput, Questionnaire, TextArea)
- Add ExternalReplyContainer component
- Add ExplanationSection component
- Add Location component
- Add useExternalReplyQuestionnaire hook

## Design

InVision designs can be found [here](https://projects.invisionapp.com/share/RF131I9Q6DVM#/screens/470029477)

## Testing this branch

- Follow instructions of https://github.com/Amsterdam/signals-frontend/pull/2414 to forward an incident to an external party
- In the [Django admin](http://localhost:8000/signals/admin/email_integrations/emailtemplate/), add the below email template for the key `Send mail signal forwarded to external reaction  received`. This example corresponds to the design but any municipality can choose their own template
- Click on the link in the [e-mail](http://localhost:8025/#) to open the questionnaire

```
Geachte behandelaar,

Bedankt voor het invullen van het actieformulier. Uw informatie help ons bij het verwerken van de melding.

**U liet ons het volgende weten**
{{ reaction }}

**Gegevens van de melding**\
Nummer: {{ formatted_signal_id }}\
Gemeld op: {{ created_at|date:"j F y, h.m" }} uur\
Plaats: {{ address  }}

Met vriendelijke groet,\
{{ ORGANIZATION_NAME }}
```

### CODE REVIEW

**Comment by vdegraaf** [comment]
Code looks good! Few functional comments:
- When the incident is passed forward to extern and then you add an image, the form that the external party had does not get updated. Also, when the form is submitted, it will overwrite the added image (but in the history you still see it is added).
- Images appear in the opposite order in the form on the external party side. Might cause confusion.
- Page does not have a footer. The design does.

File: src/signals/incident/containers/ExternalReplyContainer/__test__/ExternalReplyContainer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+import { render, waitFor, screen, act } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { mocked } from 'jest-mock'
+import * as reactRouterDom from 'react-router-dom'
+
+import * as actions from 'containers/App/actions'
+import { withAppContext } from 'test/utils'
+
+import ExternalReplyContainer from '..'
+import * as API from '../../../../../../internals/testing/api'
```
Comment (OscarBakker) [inline src/signals/incident/containers/ExternalReplyContainer/__test__/ExternalReplyContainer.test.tsx:12]: might be cleaner to add an alias in web pack's resolve config for internals. not really a blocker

File: src/signals/incident/containers/ExternalReplyContainer/components/Location/Location.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  margin-right: ${themeSpacing(2)};
+`
+
+export const MapThumbnail = styled(MapDetail).attrs({
+  canFocusMarker: false,
+  hasZoomControls: false,
+  icon: smallMarkerIcon,
+  zoom: 9,
+})`
+  width: 80px;
```
Comment (OscarBakker) [inline src/signals/incident/containers/ExternalReplyContainer/components/Location/Location.tsx:25]: no themespacing needed?

**Review by OscarBakker — COMMENTED** [review]
Small comments, proper stuff!

**Review by vdegraaf — APPROVED** [review]
Fixed my comments, so you have my blessing!

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**